handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -68,12 +68,27 @@
                     // libraries will be available if using a tcnative uber jar.
                     NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
                 } catch (Throwable t) {
-                    // Either the uber jar isn't on classpath, or it doesn't contain a library
-                    // for this platform. Load the default library.
-                    logger.debug("Unable to load platform-specific netty-tcnative library: " +
-                        libName + ". Will attempt to load the default netty-tcnative library.", t);
+                    // Either the uber jar isn't on classpath, the platform is fedora,
+                    // or it doesn't contain a library for this platform. If the OS is
+                    // Linux try fedora otherwise load the default library.
+                    boolean loaded = false;
+                    if (os.equalsIgnoreCase("linux")) {
+                        logger.debug("Unable to load platform-specific netty-tcnative library: " +
+                                     libName + ". Will attempt to load the fedora netty-tcnative library.", t);
+                        libName += "-fedora";
+                        try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)